src/OpenCensus/Resource/Resource.cs

```diff
@@ -0,0 +1,202 @@
+        /// </summary>
+        private const string OC_RESOURCE_LABELS_ENV = "OC_RESOURCE_LABELS";
+
+        /// <summary>
+        /// Tag list splitter.
+        /// </summary>
+        private const char LABEL_LIST_SPLITTER = ',';
+
+        /// <summary>
+        /// Key-value splitter.
+        /// </summary>
+        private const char LABEL_KEY_VALUE_SPLITTER = '=';
+
+        /// <summary>
+        /// Error message when string contains invalid characters.
+        /// </summary>
+        private static readonly string ERROR_MESSAGE_INVALID_CHARS =
+            " should be a ASCII string with a length greater than 0 and not exceed " + MaxResourceTypeNameLength + " characters.";
+
+        /// <summary>
+        /// Environment identification (for example, AKS/GKE/etc).
+        /// </summary>
+        private static readonly string EnvironmentType;
+
+        private static readonly ITag[] EnvironmentToLabelMap;
+
+        static Resource()
+        {
+            string openCensusResourceType;
+            string openCensusEnvironmentTags;
+
+            try
+            {
+                openCensusResourceType = Environment.GetEnvironmentVariable(OC_RESOURCE_TYPE_ENV);
+            }
+            catch (SecurityException)
+            {
+                openCensusResourceType = GlobalResourceType;
+
+                // TODO - Log error
+            }
+
+            try
+            {
+                openCensusEnvironmentTags = Environment.GetEnvironmentVariable(OC_RESOURCE_LABELS_ENV);
+            }
+            catch (SecurityException)
+            {
+                openCensusEnvironmentTags = string.Empty;
+
+                // TODO - Log error
+            }
+
+            TryParseResourceType(openCensusResourceType, out EnvironmentType);
+            EnvironmentToLabelMap = ParseResourceLabels(Environment.GetEnvironmentVariable(OC_RESOURCE_LABELS_ENV));
+        }
+
+        /// <summary>
+        /// Gets the identification of the resource.
+        /// </summary>
+        public abstract string Type { get; protected set; }
+
+        /// <summary>
+        /// Gets the map between the tag and its value.
+        /// </summary>
+        public abstract IList<ITag> Tags { get; }
+
+        /// <summary>
+        /// Creates a label/tag map from the OC_RESOURCE_LABELS environment variable.
+        /// OC_RESOURCE_LABELS: A comma-separated list of labels describing the source in more detail,
+        /// e.g. “key1=val1,key2=val2”. Domain names and paths are accepted as label keys.
+        /// Values may be quoted or unquoted in general. If a value contains whitespaces, =, or " characters, it must
+        /// always be quoted.
+        /// </summary>
+        /// <param name="rawEnvironmentTags">Environment tags as a raw, comma separated string</param>
+        /// <returns>Environment Tags as a list</returns>
+        private static ITag[] ParseResourceLabels(string rawEnvironmentTags)
```

Comment:
Can you please add some unit tests as part of this PR?

Reply:
Done